### PR TITLE
add profile stats relating to avatar CPU costs

### PR DESCRIPTION
**Pull request by AndrewMeadows**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: f0f4a06d817a52aa03de269aa3ce42b81e769b5e available at: https://gist.github.com/d12f3a97139320add15f3f166b3cc5ca<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9212-f0f4a06.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9212-f0f4a06.dmg

**Comment by AndrewMeadows** [comment]
This PR is preparation for an optimization pass on per-avatar CPU costs.  There are no optimizations here, just new stats that will be used to verify future improvements.

To test:

(1) Login to a relatively empty domain
(2) Enable the Developer menu options:  `Settings --> Developer Menus`
(3) Enable timing details: `Developer --> Timing --> Performance Timer --> Display Timing Details`
(4) Left-click with the mouse on the stats UI that shows up at the top of the screen -- this will toggle the  per-frame stats details to show up
(5) Enable extra stats:
```
Developer --> Timing --> Performance Timer --> Expand /update
Developer --> Timing --> Performance Timer --> Expand /otherAvatar
```
(6) Login to our test domain that has 100 avatars: **dev-chris.highfidelity.io**.  Note: your frame rate will probably be slow in this domain.
(7) Verify that this stat is visible (its value is not yet important):
`/idle/update/AvatarManager/otherAvatars/simulate/skeleton/Model::simulate`
(8) fin

File: interface/src/avatar/Avatar.cpp [hunk 1]
```diff
@@ -285,28 +286,38 @@ void Avatar::simulate(float deltaTime) {
...
+            qApp->copyViewFrustum(viewFrustum);
+            float visibility = calculateRenderAccuracy(viewFrustum.getPosition(),
+                    getBounds(), DependencyManager::get<LODManager>()->getOctreeSizeScale());
+            if (!_shouldAnimate) {
+                if (visibility > MINIMUM_VISIBILITY_FOR_ON) {
+                    _shouldAnimate = true;
+                    qCDebug(interfaceapp) << "Restoring" << (isMyAvatar() ? "myself" : getSessionUUID()) << "for visibility" << visibility;
+                }
+            } else if (visibility < MAXIMUM_VISIBILITY_FOR_OFF) {
+                _shouldAnimate = false;
```
Comment (sethalves) [inline interface/src/avatar/Avatar.cpp]: Is this not an actual optimization (in defiance of the test-plan)?

**Comment by claudioshg** [comment]
Passed
pr_9212

**Comment by OmegaHeron** [comment]
Wanted to see how my i5 6500 CPU + GTX970 would cope with a large AV population...

Works as outlined and I was happy to see 47 fps DK2/60 desktop.

File: interface/src/avatar/Avatar.cpp (hunk 1 above)
Comment (AndrewMeadows) [inline interface/src/avatar/Avatar.cpp]: No, nothing changed here.  This code is just getting indented for a `PerformanceTimer `context.

**Comment by hifi-gustavo** [comment]
Build console output for commit: bd2bd8d633a00bd38ed1d168119757ad7366ca61 available at: https://gist.github.com/ea4993f4ea82be06cb7e73b275cf8fa5<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9212-bd2bd8d.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9212-bd2bd8d.dmg

**Comment by AndrewMeadows** [comment]
test this please

**Comment by claudioshg** [comment]
Testing

**Comment by claudioshg** [comment]
On it 

    On Friday, December 16, 2016 1:29 PM, Andrew Meadows <notifications@github.com> wrote:
 

 test this please—

**Comment by AndrewMeadows** [comment]
Sorry @claudioshg, the "test this please" comment is an old hook to our automated build system to make it kick off another build.  It should probably be modified to be "build this please" or something like that.

**Comment by hifi-gustavo** [comment]
Build console output for commit: bd2bd8d633a00bd38ed1d168119757ad7366ca61 available at: https://gist.github.com/d12f3a97139320add15f3f166b3cc5ca<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9212-bd2bd8d.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9212-bd2bd8d.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ee6a9f45062211d8f8e1b4b0c5de12fc3e985912 available at: https://gist.github.com/d12f3a97139320add15f3f166b3cc5ca<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9212-ee6a9f4.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9212-ee6a9f4.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ee6a9f45062211d8f8e1b4b0c5de12fc3e985912 available at: https://gist.github.com/d12f3a97139320add15f3f166b3cc5ca<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9212-ee6a9f4.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9212-ee6a9f4.dmg

**Pull request merged**